xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -1387,28 +1379,17 @@ protected void handleCdsResponse(String versionInfo, List<Any> resources, String
       } catch (ResourceInvalidException e) {
         errors.add(
             "CDS response Cluster '" + clusterName + "' validation error: " + e.getMessage());
+        invalidResources.add(clusterName);
         continue;
       }
       parsedResources.put(clusterName, new ParsedResource(cdsUpdate, resource));
     }
     getLogger().log(XdsLogLevel.INFO,
         "Received CDS Response version {0} nonce {1}. Parsed resources: {2}",
         versionInfo, nonce, unpackedResources);
-
-    if (!errors.isEmpty()) {
-      handleResourcesRejected(ResourceType.CDS, unpackedResources, versionInfo, nonce, errors);
-      return;
-    }
-
-    handleResourcesAccepted(ResourceType.CDS, parsedResources, versionInfo, nonce);
-    // CDS responses represents the state of the world, EDS resources not referenced in CDS
```

Comment:
Why do we even need logic like this? If EDS is no longer needed then the subscriber will go away and we'll naturally stop watching it. I remember discussing this in the past, but maybe I don't remember the previous resolution.

Reply:
Yeah, we had a very long debate with @voidzcy one/two years ago. There was very subtle reason that convinced me to accept this approach, but I can't recall now. 

I'm keeping this behavior in this PR to implement the change, but therefore I have to add code like https://github.com/grpc/grpc-java/pull/8506/files/5fe2a6bc3dd52f2687a91903732c443567c0e6b1#diff-6b019eb1d3518381d336168847890513a235ea4974fcb3c76a2e9ace43f1d460R2036-R2052 which makes `ClientXdsClient` so strongly coupled with `LdsUpdate` and `CdsUpdate`'s deeply nested fields, or with LDS/CDS config's semantic.